### DISCUSSION TITLE
Multiple MPD continuous running

### DISCUSSION THREAD

**Discussion by shinkhell** [MusicPlayerDaemon/MPD#1438] [Q&A] [answered]
Hello!
Currently running multiple MPD sources on my Raspberry Pi 3, I use MPC's "now playing" feedback to ensure my main instance of MPD is running and playing.
For some reason, probably bad MP3 files, sometimes the source halts and I have not way to know that unless I interrogate MPC, and then via some script, switch MPD back on play mode.

As far as I know, MPD does not provide this kind of information as to its current state, and despite having set up multiple MPD sources running, I can only monitor 1 using MPC, which apparently does not take any MPD instance as argument.

Would anyone know if :
- MPD is actually able of outputting some current playing status - really, a "now playing" would suffice - via a specific argument,
- There is an equivalent to the wonderfully light command-line MPC to enquire about _any_ MPD source you would point it to?

Sorry for my noobness and thanks a bunch for this wonderful project :)

**Comment by owl770** [comment, marked as answer]
To interrogate different MPD instances, use the `--port` and/or `--host` options of `mpc`.

e.g. `mpc --port 6600` will report on the instance of MPD listening on port 6600 (the default)
        `mpc --host myPi --port 6601` will report on the instance of MPD listening on port 6601 on the host `myPi`

Using `mpc status` you can determine if MPD is playing or not.

To further investigate why MPD stops playing, you will need to 1) make sure you are running the latest version (0.23.5) and 2) produce a verbose log (either via the `mpd.conf` file or the command line when you start MPD) to see what MPD complains about.

**Comment by shinkhell** [comment]
Thank you very much for your reply! I missed the --port option, that would indeed perfectly suit my need.
As for the interruption, I also shall apply what you said and check out the logs before bothering you again :D

Thank you very much one more!
